packages/flutter_tools/lib/src/migrations/lldb_init_migration.dart

```diff
@@ -0,0 +1,146 @@
+        return;
+      }
+      _migrateScheme(schemeInfo);
+    } on Exception catch (e) {
+      logger.printError(
+        'An error occurred when adding LLDB Init File:\n'
+        '  $e',
+      );
+    }
+  }
+
+  Future<SchemeInfo> _getSchemeInfo() async {
+    final XcodeProjectInfo? projectInfo = await _xcodeProject.projectInfo();
+    if (projectInfo == null) {
+      throw Exception('Unable to get Xcode project info.');
+    }
+    if (_xcodeProject.xcodeWorkspace == null) {
+      throw Exception('Xcode workspace not found.');
+    }
+    final String? scheme = projectInfo.schemeFor(_buildInfo);
+    if (scheme == null) {
+      projectInfo.reportFlavorNotFoundAndExit();
+    }
+
+    final File schemeFile = _xcodeProject.xcodeProjectSchemeFile(scheme: scheme);
+    if (!schemeFile.existsSync()) {
+      throw Exception('Unable to get scheme file for $scheme.');
+    }
+
+    final String schemeContent = schemeFile.readAsStringSync();
+    return SchemeInfo(schemeName: scheme, schemeFile: schemeFile, schemeContent: schemeContent);
+  }
+
+  bool _isSchemeMigrated(SchemeInfo schemeInfo) {
+    if (schemeInfo.schemeContent.contains(initPath)) {
+      return true;
+    }
+
+    // If the scheme is using a LLDB Init File that is not flutter_lldbinit,
+    // print a warning to either remove their LLDB Init file or append
+    // flutter_lldbinit to their existing one. This warning will continue to
+    // show even if they add flutter_lldbinit to their lldb init file.
+    if (schemeInfo.schemeContent.contains('customLLDBInitFile') && _buildInfo.isDebug) {
+      logger.printWarning(
+        'Running Flutter in debug mode on new iOS versions requires a LLDB '
+        'Init File, but the scheme already has one set. To ensure debug '
+        'mode works, please complete one of the following:\n'
+        '  * Remove the LLDB Init File from the scheme ${schemeInfo.schemeName}\n'
+        '  * Append the following to your custom LLDB Init File:\n'
+        '      `command source ${_xcodeProject.lldbInitFile.absolute.path}`',
+      );
+      return true;
+    }
+    return false;
+  }
+
+  void _migrateScheme(SchemeInfo schemeInfo) {
+    final File schemeFile = schemeInfo.schemeFile;
+    final String schemeContent = schemeInfo.schemeContent;
+
+    final String newScheme = schemeContent.replaceAll(
+      'selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"',
+      '''
+selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      customLLDBInitFile = "$initPath"''',
+    );
+    try {
+      final XmlDocument document = XmlDocument.parse(newScheme);
+      _validateSchemeAction(action: 'LaunchAction', document: document, schemeFile: schemeFile);
+      _validateSchemeAction(action: 'TestAction', document: document, schemeFile: schemeFile);
+    } on XmlException catch (exception) {
+      throw Exception(
+        'Failed to parse ${schemeFile.basename}: Invalid xml: $newScheme\n$exception',
```

Comment:
Consider adding a little more context to this error, like "Failed to add customLLDBInitFile to Xcode scheme. Failed to..."